src/Compilers/Core/Portable/PEWriter/MetadataWriter.DynamicAnalysis.cs

```diff
@@ -0,0 +1,267 @@
+                    HashAlgorithm = (uint)(checksumAndAlgorithm.Item1.IsDefault ? 0 : _dynamicAnalysisDataHeaps.GetGuidIndex(checksumAndAlgorithm.Item2)),
+                    Hash = (checksumAndAlgorithm.Item1.IsDefault) ? new BlobIdx(0) : _dynamicAnalysisDataHeaps.GetBlobIndex(checksumAndAlgorithm.Item1)
+                });
+            }
+
+            return documentRowId;
+        }
+
+        private static readonly char[] s_separator1 = { '/' };
+        private static readonly char[] s_separator2 = { '\\' };
+
+        private BlobIdx SerializeDocumentName(string name)
+        {
+            Debug.Assert(name != null);
+
+            var writer = new BlobBuilder();
+
+            int c1 = Count(name, s_separator1[0]);
+            int c2 = Count(name, s_separator2[0]);
+            char[] separator = (c1 >= c2) ? s_separator1 : s_separator2;
+
+            writer.WriteByte((byte)separator[0]);
+
+            // TODO: avoid allocations
+            foreach (var part in name.Split(separator))
+            {
+                BlobIdx partIndex = _dynamicAnalysisDataHeaps.GetBlobIndex(ImmutableArray.Create(MetadataWriter.s_utf8Encoding.GetBytes(part)));
+                writer.WriteCompressedInteger((uint)_dynamicAnalysisDataHeaps.ResolveBlobIndex(partIndex));
+            }
+
+            return _dynamicAnalysisDataHeaps.GetBlobIndex(writer);
+        }
+
+        private static int Count(string str, char c)
+        {
+            int count = 0;
+            for (int i = 0; i < str.Length; i++)
+            {
+                if (str[i] == c)
+                {
+                    count++;
+                }
+            }
+
+            return count;
+        }
+
+        #endregion
+
+        #region Table Serialization
+
+        private sealed class Sizes
+        {
+            public readonly ImmutableArray<int> HeapSizes;
+            public readonly int BlobIndexSize;
+            public readonly int GuidIndexSize;
+
+            public Sizes(ImmutableArray<int> heapSizes)
+            {
+                HeapSizes = heapSizes;
+                BlobIndexSize = (heapSizes[(int)HeapIndex.Blob] < 0x10000) ? 2 : 4;
+                GuidIndexSize = (heapSizes[(int)HeapIndex.Guid] < 0x10000) ? 2 : 4;
+            }
+        }
+
+        internal void SerializeMetadataTables(BlobBuilder writer)
+        {
+            _dynamicAnalysisDataHeaps.Complete();
+
+            var sizes = new Sizes(_dynamicAnalysisDataHeaps.GetHeapSizes());
+
+            SerializeHeader(writer, sizes);
+
+            // tables:
+            SerializeDocumentTable(writer, sizes);
+            SerializeMethodSpanTable(writer, sizes);
+
+            // heaps:
+            int guidHeapStartOffset;
+            _dynamicAnalysisDataHeaps.WriteTo(writer, out guidHeapStartOffset);
+        }
+
+        private void SerializeHeader(BlobBuilder writer, Sizes sizes)
+        {
+            // signature:
+            writer.WriteByte((byte)'D');
```

Comment:
No. Just some 4 letters that can be easily recognized in a binary blob.  Dynamic Analysis MetaData. ;)